sacc/tracers.py

```diff
@@ -325,3 +325,45 @@ def from_tables(cls, table_list):
                     metadata[key[5:]] = value
             tracers[name] = cls(name, z, nz, extra_columns=extra_columns, metadata=metadata)
         return tracers
+
+
+def combine_dictionaries(d1, d2, suffix1=None, suffix2=None, cmp=None):
+    """Combine two dictionaries, preventing clashes and optionally renaming keys
+
+    This combines two dictionaries, raising an error in the event of
+    there being identical keys in each with different values.
+
+    This can be prevented by supplying suffix1 and/or suffix2, which will
+    be appended to the keys in each dictionary.
+
+    If there is still a clash, an error is raised.
+
+    Parameters
+    ----------
+    d1: dict
+
+    d2: dict
+
+    suffix1: str
+        Optional, default=None, suffix for keys in d1
+
+    suffix2: str
+        Optional, default=None, suffix for keys in d2
+    """
+
+    if suffix1 is not None:
+        output = {key+suffix1: value for key, value in d1.items()}
+    else:
+        output = d1.copy()
+
+    if suffix2 is not None:
+        d2 = {key+suffix1: value for key, value in d2.items()}
+
+
+    for key, value in d2.items():
+        if (key in output) and (output[key] != value):
```

Comment:
Is it a feature or a bug that two keys with the same value don't get two entries in the final dict? I feel like this will never be a problem until the one time it is. Any thoughts?

Reply:
It was going to be a feature, to avoid shared names needlessly proliferating, but I think you're right and the most conservative thing to do is require a label if there is any clash.